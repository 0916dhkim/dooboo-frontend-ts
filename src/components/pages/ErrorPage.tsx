import React from 'react';

const ErrorPage: React.StatelessComponent<{}> = () => {
  return (
    <span> Error page. Check the URL please.</span>
  );
};

export default ErrorPage;
