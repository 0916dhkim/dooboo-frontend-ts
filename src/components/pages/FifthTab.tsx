import React, { Component } from 'react';

class FifthTab extends Component {
  public render() {
    return (
      <div>Nothing</div>
    );
  }
}

export default FifthTab;
