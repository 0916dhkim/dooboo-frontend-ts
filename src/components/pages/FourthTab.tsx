import React, { Component } from 'react';

class FourthTab extends Component {
  public render() {
    return (
      <div>Nothing</div>
    );
  }
}

export default FourthTab;
