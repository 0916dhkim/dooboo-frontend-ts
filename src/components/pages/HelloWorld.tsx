import React, { Component } from 'react';

class HelloWorld extends Component<any> {
  public render() {
    return (
      <div
        onClick={() => this.props.history.goBack()}
        style={{ cursor: 'pointer', color: 'tomato'}}
      >
        GO BACK TO MAIN PAGE!
      </div>
    );
  }
}

export default HelloWorld;
