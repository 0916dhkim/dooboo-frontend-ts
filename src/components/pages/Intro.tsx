import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { inject, observer } from 'mobx-react';
import autobind from 'autobind-decorator';
import Radium from 'radium';

import { globalColors, globalEffects } from '@utils/styles';
import CooniBtn from '@shared/CooniBtn';

@autobind
@inject('store')
@observer
@Radium
class Intro extends Component<any> {

  public login() {
    this.props.store.logIn();
  }

  public signUp() {
    this.props.history.push('/signup');
  }

  public render() {
    return(
      <div>
        {
          this.props.store.loggedIn
          ? <Redirect to='/tab/tab1' />
          : <div style={styles.container}>
              <div style={styles.introContainer}>
                <div style={styles.oauthBox}>
                    <CooniBtn
                      clickHandler={() => this.signUp()}
                      white={true}
                      btnTxt='회원가입'
                    />
                    <CooniBtn
                      clickHandler={() => this.login()}
                      white={true}
                      btnTxt='구글으로 로그인'
                      imgSrc='https://icdn6.digitaltrends.com/image/google_icon-377x372.jpg'
                    />
                    <CooniBtn
                      clickHandler={() => this.login()}
                      white={true}
                      btnTxt='페이스북으로 로그인'
                      // tslint:disable-next-line:max-line-length
                      imgSrc='https://media4.s-nbcnews.com/j/streams/2013/december/131203/2d9840012-facebooklogo.nbcnews-fp-360-360.jpg'
                    />
                </div>
              </div>
            </div>
        }
      </div>
    );
  }
}

const styles: any = {
  container: {
    position: 'absolute',
    width: '100%',
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    background: globalColors.cooniGradient,
  },

  introContainer: {
    margin: '28px',
    width: 'calc(100%-56px)',
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    position: 'relative',
  },

  oauthBox: {
    bottom: 0,
    position: 'absolute',
  },

  oauthBtn: {
    color: '#fff',
    width: '320px',
    height: '28px',
    borderRadius: '3px',
    border: 'solid 2px #fff',
    margin: '8px 0',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',

    ...globalEffects.btnEffect,

    ':hover': {
      ...globalEffects.btnEffectHover,
    },
    ':active': {
      opacity: 0.5,
    },

    '@media (max-width: 740px)': {
      width: '85vw',
      height: '8vh',
    },
  },

  googleLogoImg: {
    position: 'absolute',
    left: '16px',
    height: '20px',
    width: '20px',
    objectFit: 'cover',
  },

  fbLogoImg: {
    position: 'absolute',
    left: '16px',
    height: '20px',
    width: '20px',
    objectFit: 'cover',
  },
};

export default Intro;
