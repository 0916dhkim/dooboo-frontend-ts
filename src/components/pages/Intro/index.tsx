import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { inject, observer } from 'mobx-react';

import { colors, effects } from '@utils/styles';
import CooniBtn from '@shared/CooniBtn';
import {
  google_logo_1x, google_logo_2x, google_logo_3x,
  facebook_logo_1x, facebook_logo_2x, facebook_logo_3x,
} from '@utils/Icons';

const styles = require('./style.css');

@inject('store')
@observer
class Intro extends Component<any> {
  public login(oauth: string) {
    this.props.store.user.login(`user from ${oauth}`);
  }

  public goToSignUp() {
    this.props.history.push('/signup');
  }

  public render() {
    const { getString } = this.props.store.locale;
    return(
      <div>
        {
          this.props.store.user.loggedIn
          ? <Redirect to='/tab/tab1' />
          : <div className={styles.container}>
              <div className={styles.introContainer}>
                <div className={styles.oauthBox}>
                    <CooniBtn
                      onClick={() => this.goToSignUp()}
                      white={true}
                      btnTxt={getString('SIGNUP')}
                    />
                    <CooniBtn
                      onClick={() => this.login('google')}
                      white={true}
                      btnTxt={getString('GOOGLE_LOGIN')}
                      imgSrc={google_logo_1x}
                      srcset={`${google_logo_1x} 1x, ${google_logo_2x} 2x, ${google_logo_3x} 3x`}
                    />
                    <CooniBtn
                      onClick={() => this.login('facebook')}
                      white={true}
                      btnTxt={getString('FACEBOOK_LOGIN')}
                      imgSrc={facebook_logo_2x}
                      srcset={`${facebook_logo_1x} 1x, ${facebook_logo_2x} 2x, ${facebook_logo_3x} 3x`}
                    />
                </div>
              </div>
            </div>
        }
      </div>
    );
  }
}

export default Intro;
