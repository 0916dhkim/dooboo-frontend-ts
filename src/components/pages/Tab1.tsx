import React, { Component } from 'react';
import Radium from 'radium';

const styles: any = {
  hyperLink: {
    textDecoration: 'none',
    color: 'rgb(13, 157, 197)',
    ':hover': {
      color: '#fff',
      backgroundColor: 'rgb(13, 157, 197)',
    },
  },
};

@Radium
class Tab1 extends Component<any> {
  public render() {
    return (
      <div>
        <div style={{fontSize: '20px'}}>
          boiler plate of <span style={{fontWeight: 'bold'}}>React</span> with
        </div><br/>
        <a key={0} style={styles.hyperLink} href='https://github.com/ReactTraining/react-router'>
          react-router v4
        </a><br/>
        <a key={1} style={styles.hyperLink} href='https://www.typescriptlang.org/docs/home.html'>
          typescript
        </a><br/>
        <a key={2} style={styles.hyperLink} href='https://github.com/mobxjs/mobx'>
          mobx
        </a><br/>
        <a key={3} style={styles.hyperLink} href='https://github.com/FormidableLabs/radium'>
          modular style with radium
        </a><br/>
        <a key={4} style={styles.hyperLink}
          href='https://github.com/rofrischmann/inline-style-prefixer/commit/bc71fd88a96497164e5e1f4a291301722428f780'>
          inline-style-prefixer
        </a><br/>
        <a key={5} style={styles.hyperLink} href='https://github.com/andreypopp/autobind-decorator'>
          autobind-decorator
        </a><br/>
      </div>
    );
  }
}

export default Tab1;
