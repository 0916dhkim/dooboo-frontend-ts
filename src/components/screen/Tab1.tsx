import { inject, observer } from 'mobx-react';
import React, { Component } from 'react';
import { Route, Link } from 'react-router-dom';
import CooniBtn from '@shared/CooniBtn';

const styles = require('./Tab1.css');

export class Tab1 extends Component<any> {
  public render() {
    const { getString } = this.props.store.locale;
    return (
      <div className={styles.container}>
        <div style={{ fontSize: '20px', marginTop: '30px' }}>
          boilerplate of <span style={{ fontWeight: 'bold' }}>React</span> with
        </div><br/>
        <a key={0} className={styles.hyperLink} href='https://github.com/ReactTraining/react-router'>
          react-router v4
        </a><br/>
        <a key={1} className={styles.hyperLink} href='https://www.typescriptlang.org/docs/home.html'>
          typescript
        </a><br/>
        <a key={2} className={styles.hyperLink} href='https://github.com/mobxjs/mobx'>
          mobx
        </a><br/>
        <a key={3} className={styles.hyperLink}
          href='https://github.com/rofrischmann/inline-style-prefixer/commit/bc71fd88a96497164e5e1f4a291301722428f780'>
          inline-style-prefixer
        </a><br/>
        <br/>
        <CooniBtn
          onClick={() => this.props.history.push('/none_tab')}
          btnTxt='none tab page'
        />
        <CooniBtn
          onClick={() => this.props.store.user.logout()}
          btnTxt={getString('LOGOUT')}
        />
      </div>
    );
  }
}

const defaultTab1 = inject('store')(observer(Tab1));
export default defaultTab1;
