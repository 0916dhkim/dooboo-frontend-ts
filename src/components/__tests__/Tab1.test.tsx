import React from 'react';
import { shallow, render } from 'enzyme';
import { StyleRoot } from 'radium';

import Store from '@stores/appStore';
import { Tab1 } from '../pages/Tab1';

// test for the container page in dom
describe('Tab1 page DOM rendering test', () => {
  const testStore = new Store();
  const page = shallow(
    <Tab1 test={true} store={testStore} />,
  );

  it('Tab1 page has to be matched to snapshot', () => {
    expect(page).toMatchSnapshot();
  });
});
