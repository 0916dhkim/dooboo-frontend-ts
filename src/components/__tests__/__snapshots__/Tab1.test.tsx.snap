// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab1 page DOM rendering test Tab1 page has to be matched to snapshot 1`] = `
<div>
  <div
    style={
      Object {
        "fontSize": "20px",
        "marginTop": "30px",
      }
    }
  >
    boilerplate of 
    <span
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      React
    </span>
     with
  </div>
  <br />
  <a
    href="https://github.com/ReactTraining/react-router"
    key="0"
  >
    react-router v4
  </a>
  <br />
  <a
    href="https://www.typescriptlang.org/docs/home.html"
    key="1"
  >
    typescript
  </a>
  <br />
  <a
    href="https://github.com/mobxjs/mobx"
    key="2"
  >
    mobx
  </a>
  <br />
  <a
    href="https://github.com/FormidableLabs/radium"
    key="3"
  >
    modular style with radium
  </a>
  <br />
  <a
    href="https://github.com/rofrischmann/inline-style-prefixer/commit/bc71fd88a96497164e5e1f4a291301722428f780"
    key="4"
  >
    inline-style-prefixer
  </a>
  <br />
  <br />
  <CooniBtn
    btnTxt="none tab page"
  />
  <CooniBtn
    btnTxt="로그아웃"
    onClick={[Function]}
  />
</div>
`;
